src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java

```diff
@@ -0,0 +1,182 @@
+            redis.setHost(descriptor.getHost());
+            redis.setPort(descriptor.getPort());
+            redis.setKey(descriptor.getKey());
+            redis.setPassword(descriptor.getPassword());
+            logstashIndexer = redis;
+            break;
+          case ELASTICSEARCH:
+            LOGGER.log(Level.INFO, "Migrating logstash configuration for Elastic Search");
+            URI uri;
+            try
+            {
+              uri = (new URIBuilder(descriptor.getHost()))
+                  .setPort(descriptor.getPort())
+                  .setPath("/" + descriptor.getKey()).build();
+              ElasticSearch es = new ElasticSearch();
+              es.setUrl(uri.toURL());
+              es.setUsername(descriptor.getUsername());
+              es.setPassword(descriptor.getPassword());
+              logstashIndexer = es;
+            }
+            catch (URISyntaxException | MalformedURLException e)
+            {
+              LOGGER.log(Level.INFO, "Migrating logstash configuration for Elastic Search failed: " + e.toString());
+            }
+            break;
+          case RABBIT_MQ:
+            LOGGER.log(Level.INFO, "Migrating logstash configuration for  RabbitMQ");
+            RabbitMq rabbitMq = new RabbitMq();
+            rabbitMq.setHost(descriptor.getHost());
+            rabbitMq.setPort(descriptor.getPort());
+            rabbitMq.setQueue(descriptor.getKey());
+            rabbitMq.setUsername(descriptor.getUsername());
+            rabbitMq.setPassword(descriptor.getPassword());
+            logstashIndexer = rabbitMq;
+            break;
+          case SYSLOG:
+            LOGGER.log(Level.INFO, "Migrating logstash configuration for  SYSLOG");
+            Syslog syslog = new Syslog();
+            syslog.setHost(descriptor.getHost());
+            syslog.setPort(descriptor.getPort());
+            syslog.setSyslogProtocol(descriptor.getSyslogProtocol());
+            switch (descriptor.getSyslogFormat())
+            {
+              case RFC3164:
+                syslog.setMessageFormat(MessageFormat.RFC_3164);
+                break;
+              case RFC5424:
+                syslog.setMessageFormat(MessageFormat.RFC_5424);
+                break;
+              default:
+                syslog.setMessageFormat(MessageFormat.RFC_3164);
+                break;
+            }
+            logstashIndexer = syslog;
+            break;
+          default:
+            LOGGER.log(Level.INFO, "unknown logstash Indexer type: " + type);
+            break;
+        }
+        activeIndexer = logstashIndexer;
+      }
+      dataMigrated = true;
+      save();
+    }
+  }
+
+  @Override
+  public boolean configure(StaplerRequest staplerRequest, JSONObject json) throws FormException
+  {
+    // when we bind the stapler request we get a new instance of logstashIndexer
+    staplerRequest.bindJSON(this, json);
+    if (!Objects.equals(logstashIndexer, activeIndexer))
```

Comment:
can you add further comment explaining that we check for equality because `Indexer` instance is a holder for Dao